lib/services/presets.go

```diff
@@ -158,3 +159,49 @@ func NewPresetAuditorRole() types.Role {
 	role.SetLogins(types.Allow, []string{"no-login-" + uuid.New().String()})
 	return role
 }
+
+// DefaultAllowRules has the Allow rules that should be set as default when they were no explicitly defined.
+// This is used to update the current cluster roles when deploying a new resource.
+func DefaultAllowRules() map[string][]types.Rule {
+	return map[string][]types.Rule{
+		teleport.PresetEditorRoleName: {
+			types.NewRule(types.KindConnectionDiagnostic, RW()),
+		},
+	}
+}
+
+// AddDefaultAllowRules returns a copy of the received role with a new default allow rule.
+// If the current role already defined one of the resources (either allowing or denying),
+// the role will be returned without setting the default.
+func AddDefaultAllowRules(role types.Role) types.Role {
+	defaultRules, ok := DefaultAllowRules()[role.GetName()]
+	if !ok || len(defaultRules) == 0 {
+		return role
+	}
+
+	combined := append(role.GetRules(types.Allow), role.GetRules(types.Deny)...)
+
+	for _, defaultRule := range defaultRules {
+		if resourceBelongsToRules(combined, defaultRule.Resources) {
+			continue
+		}
+
+		rules := role.GetRules(types.Allow)
+		rules = append(rules, defaultRule)
```

Comment:
can we add a debug log when we add a rule, in case someone keeps deleting the rule and it gets re-added they might get really confused

Reply:
Something like this?

![image](https://user-images.githubusercontent.com/689271/189314544-f8dfd3a9-3d00-4cae-bc51-682a0c284657.png)